### PR TITLE
Remove and readd onMapChange in onStart/onStop

### LINKED ISSUE #363: Remove map change listener when locationlayer isn't being used.

### PR DESCRIPTION
**Pull request by cammace**
Closes #363 

When the location layer plugin's disabled, we remove the onMapChange listener.

### CODE REVIEW

File: plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java [hunk 1]
```diff
@@ -489,6 +490,7 @@ public void onStop() {
     if (mapboxMap != null) {
       mapboxMap.removeOnCameraMoveListener(onCameraMoveListener);
     }
+    mapView.removeOnMapChangedListener(onMapChangedListener);
```
Comment (LukasPaczos) [inline plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: Won't this one unregister the listener during style switch resulting in disabling the location layer? See https://github.com/mapbox/mapbox-plugins-android/blob/86a67081e4a188433cfd7f51740c9dfa2fafc43e/plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java#L632-L641.

Comment (tobrun) [reply plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: that only happens with onMapReady callbacks not with end user provided map change listeners

File: plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java [hunk 2]
```diff
@@ -601,13 +612,13 @@ public void onStaleStateChange(boolean isStale) {
     @Override
     public void onMapChanged(int change) {
       if (change == MapView.WILL_START_LOADING_MAP) {
-        onStop();
+        onLocationLayerStop();
       } else if (change == MapView.DID_FINISH_LOADING_STYLE) {
         locationLayer.initializeComponents(options);
         locationLayerCamera.initializeOptions(options);
         setRenderMode(locationLayer.getRenderMode());
```
Comment (LukasPaczos) [inline plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: If the plugin is disabled and we change the style this line will draw the puck in the last location even though the plugin is disabled.

Comment (cammace) [reply plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: Yes, want to make this clear that this is a bug that is in master though, not something I introduced in this PR...

Comment (cammace) [reply plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: After looking into this, I'm not finding a solution to this fix. Will open a ticket and revisit in a separate PR.

Comment (cammace) [reply plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: https://github.com/mapbox/mapbox-plugins-android/issues/515

**Pull request merged**